### DISCUSSION TITLE
using nerdctl in Go exec.Command is unresponsive

### DISCUSSION THREAD

**Discussion by tianyax** [lima-vm/lima#1454] [Q&A] [answered]
### Description

When I call `nerdctl` in the go program to create a container, there is no response, but the container has been pulled up normally, and the `nerdctl run` process in the virtual machine has been executed. 
![image](https://user-images.githubusercontent.com/12004412/229444544-2f193233-6d1b-435c-b1f7-0da43d39ac8a.png)

```go
func RunCmdAndWait(name string, args ...string) (string, error) {
    cmd := exec.Command(name, args...)
    stdout, _ := cmd.StdoutPipe()
    stderr, _ := cmd.StderrPipe()
    if err := cmd.Start(); err != nil {
        return "", err
    }
    resp, err := io.ReadAll(stdout)
    if err != nil {
        return "", err
    }
    errB, err := io.ReadAll(stderr)
    if err != nil {
        return "", nil
    }
    err = cmd.Wait()
    ...
}

func main() {
    resp, err := RunCmdAndWait("nerdctl", "run", "-d", "--name", "redis", "redis")
    ...
}
```
This is only a problem if I need to pull the image.

#### System information:
os: macOS
lima version: 0.15.0
runtime: containerd

**Comment by AkihiroSuda** [comment, marked as answer]
This is expected, as `io.ReadAlll(stdout)` stucks until stdout is closed

You may just want this:
```
var stdout bytes.Buffer
cmd.Stdout = &stdout
```

**Reply by tianyax** [reply]
Yes, there is no problem with `cmd.Stdout = &stdout`, but I think this may be bypassing the problem? 😅 `cmd.Stdout = &stdout` spawns a goroutine and it will live until the program exits, also, I would like to ask why stdout is not closed, the nerdctl program in lima has already exited. 😄️
